go-controller/pkg/ovn/obj_retry.go

```diff
@@ -1514,14 +1514,6 @@ func (oc *Controller) WatchResource(objectsToRetry *retryObjs) (*factory.Handler
 					return
 				}
 
-				// skip the whole update if the new object doesn't exist anymore in the API server
-				newer, err = oc.getResourceFromInformerCache(objectsToRetry.oType, newKey)
```

Comment:
either ways I still think for update we shouldn't refetch? Since we are essentially not trying to do the actual update that came in L1490 on which `areEqual` was applied to check if update is needed. When we refetch we nullify the areEqual or not check since obj has changed. If the obj doesn't exist in api server delete event will take care of this right i.e stopping current updates?

Reply:
Ah this is true. If we do keep the lookup on the API server, we should definitely do it _before_ comparing old and new objects.